LDAR_Sim/model_code/ldar_sim_main.py

```diff
@@ -24,79 +24,90 @@
 from ldar_sim_run import ldar_sim_run
 import pandas as pd
 import os
+import sys
 import datetime
 import warnings
 import multiprocessing as mp
-from generic_functions import check_ERA5_file
+from generic_functions import check_ERA5_file, read_parameter_file
 
 if __name__ == '__main__':
     # ------------------------------------------------------------------------------
     # -----------------------------Global parameters--------------------------------
+    parameter_files = sys.argv[1:]
+    if len(parameter_files) == 0:
+        print('No parameter files supplied? Parameter files must be supplied as arguments')
+        print('Loading default parameters: default_parameters.yaml')
+        parameter_files = ['..//inputs_template//default_parameters.yaml']
+
+    parameters = {}
+    for parameter_file in parameter_files:
+        parameters.update(read_parameter_file(parameter_file))
+
+    # Set internal working directories
     src_dir_path = Path(os.path.dirname(os.path.realpath(__file__)))
     src_dir = str(src_dir_path)
     root_dir = str(src_dir_path.parent)
-    wd = os.path.abspath(root_dir) + "/inputs_template/"
-    output_directory = os.path.abspath(root_dir) + "/outputs/"
-    # Programs to compare; Position one should be the reference program (P_ref)
-    program_list = ['P_ref', 'P_base', 'P_alt', 'P_alt2', 'P_cont']
+    parameters['wd'] = os.path.abspath(parameters['wd']) + '//'
+    parameters['output_directory'] = os.path.abspath(parameters['output_directory']) + '//'
 
     # -----------------------------Set up programs----------------------------------
-    programs = []
+    warnings.filterwarnings('ignore')  # Temporarily mute warnings
+    for program in parameters['programs']:
+        for parameter_file in parameters['programs'][program]['parameter_files']:
+            # Accumulate parameters for each program
+            if not os.path.isabs(parameter_file):
+                parameter_file = os.path.join(parameters['wd'], parameter_file)
 
-    warnings.filterwarnings('ignore')    # Temporarily mute warnings
+            parameters['programs'][program].update(read_parameter_file(parameter_file))
 
-    for p in range(len(program_list)):
-        file = wd + program_list[p] + '.txt'
-        exec(open(file).read())
-        programs.append(eval(program_list[p]))
+        # Accumulate methods from method files
+        if 'method_files' in parameters['programs'][program]:
+            for method_file in parameters['programs'][program]['method_files']:
+                if not os.path.isabs(method_file):
+                    method_file = os.path.join (parameters['wd'], method_file)
```

Comment:
I know we haven't started enforcing but enable the PEP8 linter / autoformatter

Reply:
Thanks! Fixed.